cai_causal_graph/causal_graph.py

```diff
@@ -1726,6 +1726,49 @@ def get_common_descendants(self, node_1: NodeLike, node_2: NodeLike) -> Set[str]
 
         return self.get_descendants(node_1).intersection(self.get_descendants(node_2))
 
+    def get_nodes_between(self, node_1: NodeLike, node_2: NodeLike) -> Set[Node]:
+        """
+        Get the set of nodes that are on causal paths between two nodes.
+
+        The nodes themselves will be included in the final set.
+
+        If there are no causal paths between the nodes then an empty set will be returned.
```

Comment:
I would note here something about comparing this to get causal paths method? Mention that this is faster, but does not return all _paths_?